src/nodes/chunk_append/planner.c

```diff
@@ -226,15 +229,30 @@ ts_chunk_append_plan_create(PlannerInfo *root, RelOptInfo *rel, CustomPath *path
 				merge_plan->sortOperators = sortOperators;
 				merge_plan->collations = collations;
 				merge_plan->nullsFirst = nullsFirst;
+				bool partial_chunks = false;
+
+				/* to determine if we have partial chunks */
+				if (list_length(merge_path->subpaths) == 2)
+				{
+					Path *child1 = (Path *) linitial(merge_path->subpaths);
+					Path *child2 = (Path *) lsecond(merge_path->subpaths);
+					if (child1->parent->relid == child2->parent->relid)
+						partial_chunks = true;
+				}
```

Comment:
This would only detect it correctly if the first chunk is partial but not any other. I dont think we need this detection anyway if we keep track of the current group through relid.

Reply:
I thought that way of detecting partial chunks was not wrong, because at this stage these mergeAppend paths have been created by us, and for partially compressed chunks the mergeAppend node wraps exactly two paths, one for each part of the chunk. But I have updated this bit according to your comment.